docs/namespaces.md

```diff
@@ -0,0 +1,126 @@
+Clarifying the different endpoints, involved services and path transformations creates a common vocabulary. This page serves as an  example and forms the glossary for paths, namespaces and related concepts.
+
+## Endpoints
+
+OCIS implements the two webdav endpoints that are present in ownCloud 10:
+- `/dav/files/<userid>/` which presents the tree of the user with the given user id.
+- `/webdav/` which presents the tree of the currently logged in user. It [does not allow referencing another users files](https://github.com/owncloud/core/issues/12504#issuecomment-65259957) and only exists for [backwards compatability](https://github.com/owncloud/core/issues/12543).
+
+For backwards compatapility you can add the old `remote.php`, so these also work:
+- `/remote.php/dav/files/<userid>/`
+- `/remote.php/webdav/`
+
+## Paths
+
+Whatever follows the two endpoints is the *relative path* of a file in a users home storage. These two URLs actually point to the same file for the user with the id `u-u-i-d`:
+- `https://cloud.example.org/dav/files/<userid>/path/to/file.md`
+- `https://cloud.example.org/webdav/path/to/file.md`
+
+Since `path/to/file.md` is only relative to the users home storage we need to prefix it with a path to the users home storage in the CS3 namespace.
+
+## Services
+
+For WebDAV requests:
+
+`client -webdav-> proxy -webdav-> ocdav -cs3-> reva gateway -cs3-> reva storage provider`
+
+GET and PUT for WebDAV use the data providers instead of storage providers. For a resumable upload the [TUS protocol](https://tus.io/) can be used.
+
+Up and downloads take a different route so they can be scaled independently:
+
+`client -tus-> proxy -tus-> ocdav -tus-> reva data gateway -tus-> data provider`
+
+## Namespaces
+
+url / ocdav endpoint -> cs3 path
+
+`remote.php/webdav/path/to/file.txt` -> `/home/path/to/file.txt`
+ocdav service:
+WEBDAV_NAMESPACE=`/home/`
+                                        storage provider
+                                            driver: `eoshome`
+                                            id: `1284d238-aa92-42ce-bdc4-0b0000009154`
+                                            layout: `{{substr 0 1 .Id.OpaqueId}}/{{.Id.OpaqueId}}`
+                                                    `f/f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c`
+                                            eos namespace (in eos): `/eos/dockertest/reva/users`
+                                            mount point `/home`
+                                            full internal path: namespace + layout + relative path
+                                                                `/eos/dockertest/reva/users` + `f/f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c` + `path/to/file.txt`
+
+`remote.php/dav/files/f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c/path/to/file.txt` -> `/eos/f7fbf8c8-139b-4376-b307-cf0a8c2d0d9c/path/to/file.txt`
+ocdav service:
+DAV_FILES_NAMESPACE=`/eos/`
+                                                                storage provider
```

Comment:
I don't think we need to indent that far. I remember you indented back when we discussed this to signify that this is deeper but putting it visually this way doesn't convey that information, and I'm not sure that the deepness is necessary for understanding here.

I suggest removing the deep indent and making the storage providers on the same level